### ISSUE TITLE
Add pipeline support to REST API for bulk upsert

### ISSUE THREAD

**Issue by PhaedrusTheGreek** [elastic/elasticsearch#25601]
As discussed [here](https://github.com/elastic/elasticsearch/issues/21903), the Java API supports specifying a pipeline in a bulk upsert, but the REST API either does not (or is not documented).

```
PUT _ingest/pipeline/timestamps
{
  "description": "_timestamp",
  "processors": [
    {
      "set": {
        "field": "_source.timestamp_created",
        "value": "{{_ingest.timestamp}}"
      }
    }
  ]
}

POST my_index/test_type/_bulk?pipeline=timestamps
{"index":{"_id":"1"}}
{"field1":"val1", "counter":0}
{"update":{"_id":"2"}}
{"script":{"inline":"ctx._source.counter++;"},"upsert":{"field1":"upserted_val", "counter":0}}
```

The upserted record didn't hit the pipeline:

```
{
   "took": 1,
   "timed_out": false,
   "_shards": {
      "total": 5,
      "successful": 5,
      "failed": 0
   },
   "hits": {
      "total": 2,
      "max_score": 1,
      "hits": [
         {
            "_index": "my_index",
            "_type": "test_type",
            "_id": "2",
            "_score": 1,
            "_source": {
               "field1": "upserted_val",
               "counter": 1
            }
         },
         {
            "_index": "my_index",
            "_type": "test_type",
            "_id": "1",
            "_score": 1,
            "_source": {
               "timestamp_created": "Fri Jul 07 10:47:07 EDT 2017",
               "field1": "val1",
               "counter": 0
            }
         }
      ]
   }
}
```

**Comment by pereirapleandro** [comment]
Please, we have news about this issue?
We would like use this feature on our application 😊 

Thank's!

**Comment by original-brownbear** [comment]
Gave this a go in https://github.com/elastic/elasticsearch/pull/27075 :)

**Comment by original-brownbear** [comment]
fixed in `master` via #27075
